ngraph/core/src/op/util/arithmetic_reduction.cpp

```diff
@@ -51,7 +49,11 @@ const AxisSet op::util::ArithmeticReduction::get_reduction_axes() const
     AxisSet axes;
     if (auto const_op = as_type<op::Constant>(input_value(1).get_node()))
     {
-        axes = const_op->get_axis_set_val();
+        const auto const_data = const_op->cast_vector<int64_t>();
+        const auto input_data_rank = get_input_partial_shape(0).rank();
+        const auto normalized_axes =
+            ngraph::normalize_axes(get_friendly_name(), const_data, input_data_rank);
```

Comment:
This call will throw if the `input_data_rank` is dynamic. Without this normalization attempt the `get_axis_set_val()` was returning zeros for each negative axis stored in the constant though, which caused incorrect behavior in the tests. Is anyone aware of a situation, use case or a model for which this new approach can be problematic?

Reply:
Just additional note, as I can see `normalize_axes` calls `normalize_axis` and it won't throw for dynamic `input_data_rank` if there is no `axis < 0` in the `axes` vector, so if normalization is no needed.https://github.com/openvinotoolkit/openvino/blob/5cc7314f75727ed75a1cffe74625ba7ad745210f/ngraph/core/src/validation_util.cpp#L874-L890